### PR TITLE
doc: cleanup xsd-derived config doc text

### PR DESCRIPTION
**Pull request by dbkinder**
Start cleaning up formatting and content layout issues in the
xsd-derived configuration option documentation.  Includes adding
documentation for unnamed embedded simple types within an element (and
updates to the XSLT transformation to display these), cleanup of element
and type documentation, typos and description clarity.

Tracked-On: #5692

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

**Comment by dbkinder** [comment]
@wenlingz @NanlinXie @junjiemao1 How do I see what checks the MISRA-C check does, and how do I clear the failure?

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

**Comment by wenlingz** [comment]
> @wenlingz @NanlinXie @junjiemao1 How do I see what checks the MISRA-C check does, and how do I clear the failure?



> @wenlingz @NanlinXie @junjiemao1 How do I see what checks the MISRA-C check does, and how do I clear the failure?

@dbkinder sorry, because the current rule is: MISRA-C check is fail if don't run it. Next we will setup auto-run.

File: doc/scripts/configdoc.xsl [hunk 1]
```diff
@@ -170,35 +173,57 @@
...
+    <xsl:param name="indent"/>
+    <!-- append an (Optional) annotation if minOccurs=0) -->
+    <xsl:variable name="optional">
+      <xsl:choose>
+        <xsl:when test="@minOccurs">
+          <xsl:if test="@minOccurs = 0">
+             <xsl:text> *(Optional)*</xsl:text>
+          </xsl:if>
+        </xsl:when>
+        <!-- could show (Required) here instead, but it seemed to noisy -->
```
Comment (gvancuts) [inline doc/scripts/configdoc.xsl:203]:
```suggestion
        <!-- could show (Required) here instead, but it seemed too noisy -->
```

**Review by gvancuts — APPROVED** [review]

**Pull request merged**